### ISSUE TITLE
Ability to configure the different remote backend state  with ProviderConfig

### ISSUE THREAD

**Issue by linuxbsdfreak** [upbound/provider-terraform#49]
### What problem are you facing?
TF currently saves the remote state for a run as K8s secret for terraform in the cluster.  

### How could Official Terraform Provider help solve your problem?
Add the ability to save the TF state to an S3/GCP  or Azure Object store  via the ProviderConfig

```
kind: ProviderConfig
metadata:
  name: tf-provider-config
  namespace: upbound-system
spec:
  credentials:
  - filename: .git-credentials # use exactly this filename
    source: Secret
    secretRef:
      namespace: upbound-system
      name: git-credentials
      key: .git-credentials
  configuration: |
      terraform {
            backend "s3" {
               bucket         = "terraform-backend"
               key              = "tfstate"
               region         = "ap-southeast-2"
              encrypt        = true
         }
     }
```

It would nice to have the key value configurable per workload . 

If this feature is already present. It would be nice to have some examples for documentation of usage.

**Comment by bobh66** [comment]
@linuxbsdfreak - is this what you are looking for:  https://github.com/upbound/provider-terraform/blob/main/examples/providerconfig-aws.yaml#L14

**Comment by linuxbsdfreak** [comment]
No. I am looking to store the TF state to an s3 bucket or GCP or Azure blogstorage instead of the secret in K8s. 

Can i do something like this in the ProviderConfig?  Examples on usage would be helpful

```
      terraform {
        backend "s3" {
               bucket         = "terraform-backend"
               key              = "tfstate"
               region         = "ap-southeast-2"
              encrypt        = true
        }
      }
```

**Comment by bobh66** [comment]
You can put whatever terraform configuration commands you like in the ProviderConfig spec.configuration

**Comment by jeanduplessis** [comment]
⚠️ Please note that this repo has migrated to https://github.com/crossplane-contrib/provider-terraform 

Due to GitHub's limitations on transferring issues across organizations, this issue will be closed. 
Please re-open your issue upstream if you still want it considered.

Note that Upbound will no longer actively invest in this provider, but it will endeavor to continue supporting community contributions and publishing new releases as needed.

For users interested in using newer Terraform versions, please note that provider-terraform will NOT support versions of Terraform under the BUSL license (anything after 1.5.x). We encourage users who need newer capabilities to consider https://github.com/upbound/provider-opentofu.
